src/Extensions/NexusMods.Extensions.Hashing/AbsolutePathExtensions.cs

```diff
@@ -30,4 +31,45 @@ public static class AbsolutePathExtensions
             });
     }
 
+    /// <summary>
+    /// Calculates the xxHash64 of a file by memory mapping it and reports progress upon completion.
+    /// </summary>
+    /// <param name="input">The path to the file.</param>
+    /// <param name="job">The job to report progress to.</param>
+    /// <returns>The xxHash64 hash of the file.</returns>
+    public static unsafe Hash XxHash64MemoryMapped(this AbsolutePath input, IActivitySource<Size>? job = null)
+    {
+        #if DEBUG
+        // TODO: Assert that AbsolutePath uses RealFileSystem.
+        // This needs API update in paths, and an corresponding issue.
+        // If it's not RealFileSystem, we can fallback to the stream based version.
+        // Since we can't memory map, unless we add mmap ti Paths library.
+        #endif
+
+        var fullFilePath = input.GetFullPath();
+        try
+        {
+            using var memoryMappedFile = MemoryMappedFile.CreateFromFile(fullFilePath, FileMode.Open);
+            using var accessor = memoryMappedFile.CreateViewAccessor(0, 0, MemoryMappedFileAccess.Read);
+            var ptrData = (byte*)accessor.SafeMemoryMappedViewHandle.DangerousGetHandle();
+            var hashValue = XxHash64Algorithm.HashBytes(new ReadOnlySpan<byte>(ptrData, (int)accessor.Capacity));
+            job?.AddProgress(Size.FromLong(accessor.Capacity));
+            return Hash.From(hashValue);
+        }
+        catch (ArgumentException)
+        {
+            // TODO: A better way to catch this without a perf penalty.
+            // Wish there were lower level primitives for MemoryMappedFiles,
+            // but we have to live with CreateFromFile being the fastest for now.
+
+            // Empty file.
+            if (new FileInfo(fullFilePath).Length != 0) 
+                throw;
+
+            // JIT turns this to a constant.
+            // xxh64sum --little-endian "empty"
+            var emptyFileHash = BitConverter.IsLittleEndian ? 0x99e9d85137db46ef : 0xef46db3751d8e999;
+            return Hash.From(emptyFileHash);
```

Comment:
Why do you return these magic numbers instead of a zeroed hash?

Reply:
The hash function is defined as 

```csharp
public static ulong HashBytes(ReadOnlySpan<byte> data, ulong seed = 0)
{
    var hash = new XxHash64Algorithm(seed);
    var initialSize = (data.Length >> 5) << 5;
    if (initialSize > 0)
        hash.TransformByteGroupsInternal(data[..initialSize]);

    return hash.FinalizeHashValueInternal(data[initialSize..]);
}
```

The numbers are equivalent of what we'd do if we ran 
```csharp
byte dummy = 0;
`HashBytes(new Span<byte>(&dummy, 0x0))`
```

I could replace the code with that, but it won't fold to a constant.
Could move the constant to the hash library though.

Reply:
Remove the magic numbers from the code. You can make them const fields and add some comments, or move them into the hash library.